libsolidity/analysis/TypeChecker.cpp

```diff
@@ -592,6 +592,10 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 			unsigned pushes = 0;
 			if (dynamic_cast<FunctionDefinition const*>(declaration))
 				pushes = 1;
+			else if (dynamic_cast<MagicVariableDeclaration const*>(declaration))
+			{
+				return false;
```

Comment:
Wouldn't the proper fix be to put `else return false;` at the end (and / or call `fatalTypeError()`)?